### PR TITLE
Transaction loader for Buy Tokens & Claim Rewards/P1 (new) Connect Account Dashboard to new smart contracts

### PR DESCRIPTION
**Pull request by Space-zee**

### CODE REVIEW

File: api/src/features/assets/assets.controller.ts [hunk 1]
```diff
@@ -66,4 +57,28 @@ export class AssetsController {
       response.json({ resource });
     }
   }
+
+  @Post('array')
```
Comment (ruslanpalagin) [inline api/src/features/assets/assets.controller.ts:61]: sorry but this is a mess if we are trying to achieve REST API
we need to discuss it

File: frontend/app/abis/UiProvider.json [hunk 2]
```diff
@@ -0,0 +1,331 @@
+[
```
Comment (ruslanpalagin) [inline frontend/app/abis/UiProvider.json:1]: 1 why do we need so huge abi? can we put those 2 methods here?

File: frontend/app/features/account/models/accountModel.ts [hunk 3]
```diff
@@ -103,23 +98,37 @@ export const $doLoadMyOrderHistory = atom(null, async (get, set, args: { address
...
-  await waitFor(() => !!get($assetsTokenSmartContractSigned), 3)
-
-  const sc = get($assetsTokenSmartContractSigned) as AssetManager
-  const response = await sc.getMyRewardsPerAsset();
+export const $doLoadMyRewards = atom(null, async (get, set, args: { address: `0x${string}` }) => {
+  await waitFor(() => !!get($uiProviderContract), 3)
+
+  const sc = get($uiProviderContract)
+  const response = await sc.getMyRewards(args.address);
+  console.log('response', response)
```
Comment (ruslanpalagin) [inline frontend/app/features/account/models/accountModel.ts:108]: please remove logs unless they are verbose AND needed

**Review by ruslanpalagin — CHANGES_REQUESTED** [review]

**Comment by Space-zee** [comment]
https://app.clickup.com/t/860pdf480

**Comment by Space-zee** [comment]
https://app.clickup.com/t/860pbrmcv

File: frontend/deploys/localhost.json [hunk 4]
```diff
@@ -1,10 +1,14 @@
 {
-  "Usdt": "0xefAB0Beb0A557E452b398035eA964948c750b2Fd",
-  "SeriesMaster": "0x045857BDEAE7C1c7252d611eB24eB55564198b4C",
-  "KycStore": "0x071586BA1b380B00B793Cc336fe01106B0BFbE6D",
-  "AddressesProvider": "0xaca81583840B1bf2dDF6CDe824ada250C1936B4D",
-  "CoreManager": "0xA56F946D6398Dd7d9D4D9B337Cf9E0F68982ca5B",
-  "TokenPriceOracle": "0x71089Ba41e478702e1904692385Be3972B2cBf9e",
-  "RewardsDistributor": "0x02df3a3F960393F5B349E40A599FEda91a7cc1A7",
-  "UiProvider": "0xe70f935c32dA4dB13e7876795f1e175465e6458e"
+  "Usdt": "0xB1fCe46677aEF6F08c759357a4E16d020E76c562",
```
Comment (ruslanpalagin) [inline frontend/deploys/localhost.json:2]: please clean redundant addresses as it was before

**Review by ruslanpalagin — APPROVED** [review]

File: frontend/app/shared/walletsConnect/chainInfo.ts [hunk 5]
```diff
@@ -149,8 +149,8 @@ export const getActiveConfig = (preferredRpcName: string = '') :ChainInfo | null
     return null;
   }
   const rpcName = preferredRpcName || {
-    'localhost': 'arbitrumGoerli',
-    'localhost.com': 'arbitrumGoerli',
+    'localhost': 'localhost',
```
Comment (ruslanpalagin) [inline frontend/app/shared/walletsConnect/chainInfo.ts:152]: please use FF_RPC_NAME instead of changing the code

**Pull request merged**